### PR TITLE
bank: bank-with-server

### PR DESCRIPTION
**Pull request by BorislavAnchev**
The bank app made to work with Node server.

### CODE REVIEW

**Comment by BorislavAnchev** [comment]
The bank app uploaded and ready for review.

Have in mind that MongoDB Atlas' servers ocassionally have connecitvity issues.
They had connectivity problems on Tuesday and Wednesday mornings till noon, but started responding again in the afternoons. So, in general it works. The problem is not in the API.

For example, if incorrect URL in the browser, I get the correct error displayed, no matter that the MongoDB Atlas' servers are not responding.

**Comment by BorislavAnchev** [comment]
@mgenov, code ready for review. I also notice that this repository is not being 'watched' by you, so I don't know if you receive notifications for pull requests or comments here.

File: RESTful-API/api/models/accounts.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const mongoose = require('mongoose');
+
+const accountSchema = {
```
Comment (mgenov) [inline RESTful-API/api/models/accounts.js]: mongodb is not required for this task.

File: banktask/src/App.test.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  return wrapper;
+}
+
+describe('App component', () => {
+    
+  let wrapper;
+  beforeEach(() => {
+    const initialState = {
+      accounts: {
+        _u70nyuzcq: {
```
Comment (mgenov) [inline banktask/src/App.test.js]: Looks like too much testing data. 

It is more readable if testing data is in a test as it is coupled with the concrete scenario.

File: banktask/src/App.test.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import { MemoryRouter } from 'react-router-dom';
+import { mount } from 'enzyme';
```
Comment (mgenov) [inline banktask/src/App.test.js]: Consider using of react-testing-library for testing: https://github.com/testing-library/react-testing-library as it doesn't enforce coupling of specific implementation with tests.

File: banktask/src/mocks/backend.js [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+import MockAdapter from 'axios-mock-adapter';
+import { randomIdGenerator } from '../utils';
+
+const mock = new MockAdapter(axios);
+
+let database = {
+  accounts: [
+    { _id: '_u70nyuzcq', iban: 'BG12BUIN12341234567891', currency: 'BGN', balance: '5678.00', history: [
+      {
+        date: '05.01.2018',
```
Comment (mgenov) [inline banktask/src/mocks/backend.js]: no information for the user?

File: banktask/src/redux/modules/account/actions.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  }
+}
+
+export const updateAccount = (id, amount, transactionType) => {
+  return {
+    type: TYPES.UPDATE_ACCOUNT,
+    payload: {
+      request: {
+        method: 'patch',
+        url: 'http://localhost:3001/accounts',
```
Comment (mgenov) [inline banktask/src/redux/modules/account/actions.js]: relative URLs are always preferred.

**Comment by mgenov** [comment]
This PR has +20,334 changed lines and it makes it really hard for review. Please try to split it up into separate PRs. 

1. initial project setup 
2. add transaction list
3. add deploy and withdraw 
and etc

File: banktask/src/redux/modules/account/actions.js (hunk 5 above)
Comment (BorislavAnchev) [reply banktask/src/redux/modules/account/actions.js]: Changed.

**Comment by BorislavAnchev** [comment]
There is a new PR with the first part of the code - the initial project setup, but I'm not sure if you receive notification about it since you are not listed as a reviewer in that PR.

**Comment by mgenov** [comment]
Outdated.

**Pull request closed without merging**